app/Foliage/PreparePackageVersion.hs

```diff
@@ -57,27 +93,67 @@ preparePackageVersion inputDir metaFile = do
   let pkgId = PackageIdentifier pkgName pkgVersion
 
   pkgSpec <-
-    readPackageVersionSpec' (inputDir </> metaFile) >>= \case
-      PackageVersionSpec {packageVersionRevisions, packageVersionTimestamp = Nothing}
-        | not (null packageVersionRevisions) -> do
-            error $
-              unlines
-                [ inputDir </> metaFile <> " has cabal file revisions but the original package has no timestamp.",
-                  "This combination doesn't make sense. Either add a timestamp on the original package or remove the revisions"
-                ]
-      PackageVersionSpec {packageVersionRevisions, packageVersionTimestamp = Just pkgTs}
-        | any ((< pkgTs) . revisionTimestamp) packageVersionRevisions -> do
-            error $
-              unlines
-                [ inputDir </> metaFile <> " has a revision with timestamp earlier than the package itself.",
-                  "Adjust the timestamps so that all revisions come after the original package"
-                ]
-      meta ->
-        return meta
+    readPackageVersionSpec' (inputDir </> metaFile) >>= \meta@PackageVersionSpec {..} -> do
+      case (NE.nonEmpty packageVersionRevisions, packageVersionTimestamp) of
+        (Just _someRevisions, Nothing) ->
+          error $
+            unlines
+              [ inputDir </> metaFile <> " has cabal file revisions but the package has no timestamp.",
+                "This combination doesn't make sense. Either add a timestamp on the original package or remove the revisions."
```

Comment:
... is the package version timestamp optional? Surely it should not be?

Reply:
It used to be that we didn't care about index-state.

Reply:
:see_no_evil: